Adding subtyping option and fixes to -json

WilliamZekaiWang opened the pull request.

GopiGugan commented on README.md:
````diff
@@ -127,6 +127,201 @@ art@Jesry:~/git/sierra-local$ R
+```
+Note that this doesn't require any `sudo` privileges.
+
+### Graphical user interface (GUI)
+We have a GUI in development that uses the Python Tkinter framework.  To use this script, you need to check out the *gui* branch of this repository and then call the script from the root directory of the project:
+```
+python3 gui.py
+```
````
We should verify that this still works before adding it to the README

GopiGugan commented on README.md:
```diff
@@ -127,6 +127,201 @@ art@Jesry:~/git/sierra-local$ R
+- [NucAmino](https://github.com/hivdb/nucamino) `v0.1.3` or later (included with the package).
+
+[Post-Align](https://github.com/hivdb/post-align) is the new alignment program and requires the following dependency:
+- [Cython==0.29.32](https://pypi.org/project/Cython/)
```
Replace link with https://pypi.org/project/Cython/0.29.32/

GopiGugan commented on README.md:
````diff
@@ -127,6 +127,201 @@ art@Jesry:~/git/sierra-local$ R
+cd sierra-local
+sudo python3 setup.py install
+```
+Note that you need super-user privileges to install the package by this method.  For more detailed instrucitons, please refer to the document [INSTALL.md](INSTALL.md) that should be located in the root directory of this Python package.
````
Typo in instructions

GopiGugan commented:
Running into an error with post-align:

```console
Traceback (most recent call last):
  File "/usr/local/bin/postalign", line 8, in <module>
    sys.exit(cli())
  File "/home/vagrant/.local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/vagrant/.local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/vagrant/.local/lib/python3.10/site-packages/click/core.py", line 1675, in invoke
    sub_ctx = cmd.make_context(
  File "/home/vagrant/.local/lib/python3.10/site-packages/click/core.py", line 920, in make_context
    self.parse_args(ctx, args)
  File "/home/vagrant/.local/lib/python3.10/site-packages/click/core.py", line 1378, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "/home/vagrant/.local/lib/python3.10/site-packages/click/core.py", line 2360, in handle_parse_result
    value = self.process_value(ctx, value)
  File "/home/vagrant/.local/lib/python3.10/site-packages/click/core.py", line 2322, in process_value
    value = self.callback(ctx, self, value)
TypeError: Argument 'value' has incorrect type (expected list, got tuple)
```

Python version - 3.10.6
Cython - 0.29.32

ArtPoon commented:
This seems to be a postalign issue.  Please cross-post to that repo

WilliamZekaiWang commented:
List of changes

- Post-align is now the default alignment program
- Added `-subtype` which allows users to call the `subtyping.py` script automatically
- Added `-alignment` which allows users to specify choosing `nuc-amino` or `post-align`
- Added new fields to the `mutations`

   - `isApobecMutation` from file `apobecs.csv`
   - `isUnusual` from file `rx-all_subtype-all.csv`
   - `primaryType` from file `mutation-type-pairs_hiv1.csv`
   - `isSDRM` from file `sdrms_hiv1.csv`
   - `hasStop` and `text` fields

- Added `level` field to `drugScores`
- Added unittests

ArtPoon commented:
Our best guess is that this is a dependency version issue with the postalign docker container.  We might look into the click module for clues on the TypeError being raised.  For now I think we might as well merge this PR because postalign is broken in master branch anyways.
